Penumbra/UI/SettingsInterface.cs

```diff
@@ -598,36 +598,56 @@ private void DrawEditButtons()
-            var hasBottomTypes = ( _selectedMod.Mod.Meta.Groups.BottomTypes?.Count ?? 0 ) > 1;
-            var hasGroups = ( _selectedMod.Mod.Meta.Groups.OtherGroups?.Count ?? 0 ) > 1;
-            var numSelectors = ( hasTopTypes ? 1 : 0 ) + ( hasBottomTypes ? 1 : 0 ) + ( hasGroups ? 1 : 0 );
-            var selectorWidth = ( ImGui.GetWindowWidth()
-                                  - ( hasTopTypes ? ImGui.CalcTextSize( "Top       " ).X : 0 )
-                                  - ( hasBottomTypes ? ImGui.CalcTextSize( "Bottom       " ).X : 0 )
-                                  - ( hasGroups ? ImGui.CalcTextSize( "Group       " ).X : 0 ) ) / numSelectors;
-
-            void DrawSelector( string label, string propertyName, System.Collections.Generic.List< string > list, bool sameLine )
-            {
-                var current = ( int )_selectedMod.GetType().GetProperty( propertyName ).GetValue( _selectedMod );
-                ImGui.SetNextItemWidth( selectorWidth );
-                if( sameLine ) ImGui.SameLine();
-                if( ImGui.Combo( label, ref current, list.ToArray(), list.Count() ) )
-                {
-                    _selectedMod.GetType().GetProperty( propertyName ).SetValue( _selectedMod, current );
-                    _plugin.ModManager.Mods.Save();
-                    _plugin.ModManager.CalculateEffectiveFileList();
+            //Spahetti code time
+            var mod = _plugin.SettingsInterface._selectedMod;
+            var conf = mod.Conf;
+            if(conf == null) {
+                mod.Conf = new();
+                conf = mod.Conf;
+                _plugin.ModManager.Mods.Save();
+                _plugin.ModManager.CalculateEffectiveFileList();
+            }
+            var settings = mod.Mod.Meta.Groups;
+            foreach(var g in settings) {
+                if(!conf.ContainsKey(g.Key)) {
+                    conf[g.Key] = 0;
+                }
+                switch(g.Value.SelectionType) {
+                    case "Multi":
+                    {
+                        var i = 0;
+                        var flag = conf[g.Key];
+                        foreach(var opt in g.Value.Options) {
+                            var enab = (flag & 1<<i)!=0;
+                            if(ImGui.Checkbox(opt.OptionName, ref enab)) {
+                                flag = flag ^= 1<<i;
+                                conf[g.Key] = flag;
+                                _plugin.ModManager.Mods.Save();
+                                _plugin.ModManager.CalculateEffectiveFileList();
+                            }
+                            i++;
+                        }
+                        break;
+                    }
+                    case "Single": 
+                    {
+                        var code = conf[g.Key];
+                        if(ImGui.Combo(g.Value.GroupName, ref code, g.Value.Options.Select(x=>x.OptionName).ToArray(), g.Value.Options.ToArray().Length)) {
```

Comment:
I think I would skip this if Options has only one entry. Currently Tight & Firm body has a pulldown menu for 'Required Files' where you can't select anything.

Reply:
This also implicitly allows for a hidden base option for files that are included in every option. :)

Reply:
changed in c7d4cc7